src/authz/index.ts

```diff
@@ -0,0 +1,90 @@
+import axios from 'axios';
+import {FronteggAuthenticator} from "../authenticator";
+import { config } from '../config';
+import Logger from "../helpers/logger";
+import {StrategyType} from "./strategy-types";
+
+export class AuthzClient {
+
+    private readonly authenticator: FronteggAuthenticator | undefined;
+    private readonly clientId: string | undefined;
+    private strategy: IAuthzStrategy;
+
+    constructor(options: IFronteggAuthzOptions | IOpaAuthzOptions) {
+        if ('authenticator' in options) {
```

Comment:
I am not sure this is the best approach to handle it
https://stackoverflow.com/questions/12789231/class-type-check-in-typescript

Reply:
Instance of can be used only on classes not interfaces. Googling this, I've found the solution I wrote. If you have another one for interfaces, let me know